### ISSUE TITLE
Correct way to work on features for pre-release

### ISSUE THREAD

**Issue by DavidHe1127** [semantic-release/semantic-release#1510]
As per [pre-release workflow](https://github.com/semantic-release/semantic-release/blob/0785a844fa8ac1320383452ce531898be3b01f92/docs/recipes/pre-releases.md), it looks like features I will need for pre-release should be done directly on pre-release branches i.e `alpha` in the docs.

I would like to know if this is the correct way to follow?

What I have been doing  as an experimental purpose is have a dedicated feature branch, make a PR out of it and have it merged into `prerelease` branch. However, I keep getting message saying `The commit should not trigger a release` and don't know why. Below describes how I produce this message:

Both `master` and `prerelease` have the same code base and are on `1.6.1`.

- Create a `feat/a` branch out of master
- Develop the feature on `feat/a` and merge it to `prerelease`
- Publish from `prerelease` which results in `1.7.0-prerelease.1`
- Now create another `feat/b` branch out of master and do the same as `feat/a`. Keep `1.7.0-prerelease.1` when resolving merge conflicts.
- Publish from `prerelease` again

This will see the message below:
```
[11:09:31 PM] [semantic-release] › ℹ  Found git tag v1.7.0-prerelease.1 associated with version 1.7.0-prerelease.1 on branch prerelease
[11:09:31 PM] [semantic-release] › ℹ  Found 1 commits since last release
[11:09:31 PM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[11:09:31 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Feat/1.7.0.b (#17)

* fix(dave): next ver 1.6.1 (#13)

* chore(release): 1.6.1 [skip ci]

## [1.6.1](https://github.com/DavidHe1127/david-test-module/compare/v1.6.0...v1.6.1) (2020-03-31)

### Bug Fixes

* **dave:** next ver 1.6.1 ([#13](https://github.com/DavidHe1127/david-test-module/issues/13)) ([a48caa1](https://github.com/DavidHe1127/david-test-module/commit/a48caa1c8b973a7b37aa4372a8ba336261c051b6))

* feat(dave): from 1.7.0.b
```

It shows the feature change i've done on `feat/b` which I think should contribute to another prerelease. However it doesn't.  Any idea why?

Here is my config:

```
// semantic release config
// This test run was triggered on the branch chore/tidy-up, while semantic-release is configured to only publish from master, prerelease/support-prerelease-glob-name, therefore a new version won’t be published.
module.exports = {
  branches: [
    'master',
    { name: 'prerelease', channel: 'prerelease', prerelease: true },
  ],
  plugins: [
    '@semantic-release/commit-analyzer',
    '@semantic-release/release-notes-generator',
    '@semantic-release/npm',
    '@semantic-release/github',
    '@semantic-release/git',
  ],
  ci: false,
};
```

In summary, I would like to know the correct way to roll out features in pre-release.

**Comment by gr2m** [comment]
Hmm I don't know what the problem could might be. Could you run with `--debug` please, or add `debug: true` to your configuration, and paste the full logs here?

**Comment by DavidHe1127** [comment]
@gr2m Thanks for your reply. The problem lies in the way I merge which is `squash merge`. As people discussed in another [thread](https://github.com/semantic-release/semantic-release/issues/792), squash merge will prevent `commit-analyzer` `semantic-release` depends on working out whether or not a release needs to be cut. 

Once I switched my merge type to `3-way merge` everything starts working again.

I will close this issue.
